packages/core/src/lib/ui_builder.js

```diff
@@ -621,96 +613,92 @@ const ui_builder = function() {
                     logger.error('Error building ViewAllHTML');
                   });
               })
-              .then(() => {
-                //do not create a viewall page for flat patterns
-                if (!writeViewAllFile || !p) {
+              .catch(reason => {
+                console.log(reason);
+                logger.error('Error building footer HTML');
+              });
+          }
+        );
+
```

Comment:
Moved creation of view all pages down here because the files where generated multiple times. With this approach, it will only happen once for every view all page.